integration_test/cases/hibernate_test.exs

```diff
@@ -0,0 +1,46 @@
+defmodule TestIdleHibernate do
+  use ExUnit.Case, async: true
+
+  alias TestPool, as: P
+  alias TestAgent, as: A
+
+  @tag :idle_hibernate
```

Comment:
This test should have the `:idle_timeout` tag because it relies on idle timeout being active. The tags are used for test filtering, so if a pool doesn't support idle timeout it can't hibernate after an idle timeout. If there is a custom pool that needs this tag then we can leave it on. I think we should also move this test to the idle_test.exs file.